static/js/publisher/release/components/releasesConfirmDetails.js

```diff
@@ -0,0 +1,306 @@
+      </Fragment>
+    )}
+    {!progress && (
+      <Fragment>
+        <span />
+        <span />
+      </Fragment>
+    )}
+    {notes && (
+      <span className="p-releases-confirm-details__row-notes">
+        <small>{notes}</small>
+      </span>
+    )}
+  </div>
+);
+
+ReleaseRow.propTypes = {
+  type: PropTypes.string,
+  revisionInfo: PropTypes.object,
+  channel: PropTypes.node,
+  progress: PropTypes.node,
+  notes: PropTypes.node
+};
+
+const CancelProgressiveRow = ({ release }) => {
+  const revisionInfo = release.revision;
+
+  return revisionInfo.architectures.map(arch => {
+    const previousRevision = release.previousRevisions[0];
+    return (
+      <div
+        className="p-releases-confirm-details__row is-closing"
+        key={`close-${revisionInfo.revision}-${release.channel}`}
+      >
+        <span>
+          Cancel progressive release of <b>{revisionInfo.revision}</b> in{" "}
+          <b>{release.channel}</b> on <b>{arch}</b>. Revert to{" "}
+          <b>{previousRevision.revision}</b>.
+        </span>
+      </div>
+    );
+  });
+};
+
+CancelProgressiveRow.propTypes = {
+  release: PropTypes.object
+};
+
+const _ProgressiveRow = ({
+  release,
+  updateProgressiveReleasePercentage,
+  type
+}) => {
+  if (!release.progressive) {
+    return false;
+  }
+
+  let startingPercentage = 100;
+  switch (type) {
+    case progressiveTypes.RELEASE:
+      startingPercentage = release.progressive.percentage;
+      break;
+    case progressiveTypes.UPDATE:
+      startingPercentage = release.revision.release.progressive.percentage;
+      break;
+    default:
+  }
+
+  const [percentage, setPercentage] = useState(
+    release.progressive ? release.progressive.percentage : 100
+  );
+  const revisionInfo = release.revision;
+  const channel = release.channel;
+  const updatePercentage = percentage => {
+    setPercentage(percentage);
+    updateProgressiveReleasePercentage(release.progressive.key, percentage);
+  };
+
+  let progress;
+  if (
+    type === progressiveTypes.UPDATE &&
+    release.progressive.changes.some(change => change.key === "paused")
+  ) {
+    const paused = release.progressive.changes.find(
+      change => change.key === "paused"
+    ).value;
+    progress = (
+      <Fragment>
+        <ProgressiveBar percentage={percentage} />
+        <span>{paused ? "Paused" : "Resumed"}</span>
+      </Fragment>
+    );
+  } else {
+    progress = (
+      <Fragment>
+        <InteractiveProgressiveBar
+          percentage={startingPercentage}
+          onChange={updatePercentage}
+          targetPercentage={percentage}
+          minPercentage={1}
+          singleDirection={type === progressiveTypes.UPDATE ? 1 : 0}
+        />
+        <span>
+          <span className="p-tooltip--btm-right">
+            <span className="p-help">{percentage}% of devices</span>
+            <span className="p-tooltip__message">
+              Releases are delivered to devices via snap refreshes, as such, is
```

Comment:
Typo, should be `it may...` or `this may...`?
```suggestion
              Releases are delivered to devices via snap refreshes, as such, it
```